testsuite/ccp/ext5-obj-5.c

```diff
@@ -0,0 +1,11 @@
+/* { dg-options "-DCE_EXTRACT_FUNCTIONS=pu_f -DCE_EXPORT_SYMBOLS=ee_o" }*/
+static char ee_o[] = "123";
+
+void pu_f(void)
+{
+        char s[sizeof(ee_o)];
```

Comment:
can you also add two checks:
-  `char t[sizeof(ee_o)][sizeof(ee_o)];`
- The new sizeof matches the sizeof with the old declaration. You can do that by declaring an auxiliary array: `char v[sizeof(x) - 4]` for example. Any array size >= 0 will be accepted by the compiler and thus you can implement an `static_assert` with it. Or perhaps just do straight `_Static_assert()`.